test/e2e/e2ethanos/helpers.go

```diff
@@ -19,3 +23,38 @@ func CleanScenario(t *testing.T, s *e2e.Scenario) func() {
 		s.Close()
 	}
 }
+
+func singleJoiningSlash(a, b string) string {
+	aslash := strings.HasSuffix(a, "/")
+	bslash := strings.HasPrefix(b, "/")
+	switch {
+	case aslash && bslash:
+		return a + b[1:]
+	case !aslash && !bslash:
+		return a + "/" + b
+	}
+	return a + b
+}
+
+// NewSingleHostReverseProxy is almost same as httputil.NewSingleHostReverseProxy
+// but it performs a url path rewrite.
+func NewSingleHostReverseProxy(target *url.URL, externalPrefix string) *httputil.ReverseProxy {
+	targetQuery := target.RawQuery
+	director := func(req *http.Request) {
+		req.URL.Scheme = target.Scheme
+		req.URL.Host = target.Host
+		req.URL.Path = singleJoiningSlash(target.Path, strings.TrimPrefix(req.URL.Path, "/"+externalPrefix))
+
+		if targetQuery == "" || req.URL.RawQuery == "" {
+			req.URL.RawQuery = targetQuery + req.URL.RawQuery
+		} else {
+			req.URL.RawQuery = targetQuery + "&" + req.URL.RawQuery
+		}
+
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// Explicitly disable User-Agent so it's not set to default value.
```

Comment:
In understand why a general purpose library like go's net/http would need this, buy why do we need this logic?

Reply:
Gotcha, I will remove the unnecessary bits.